synapse/models/geospace.py

```diff
@@ -1,85 +1,109 @@
+import struct
+
+import synapse.exc as s_exc
 import synapse.lib.gis as s_gis
 import synapse.lib.types as s_types
+import synapse.lib.module as s_module
 import synapse.lib.syntax as s_syntax
 
-import synapse.lib.module as s_module
+units = {
+    'mm': 1,
+    'cm': 10,
+
+    'm': 1000,
+    'meters': 1000,
 
-class LatLongType(s_types.DataType):
+    'km': 1000000,
+}
+
+
+class LatLong(s_types.Type):
+    # FIXME nothing uses these props
+    # FIXME should these be indexed as something other than a string?
+
+    def postTypeInit(self):
+        self.setNormFunc(str, self._normPyStr)
 
-    def norm(self, valu, oldval=None):
+    def _normPyStr(self, valu):
```

Comment:
Probably also need a norm for tuple?

Reply:
Definitely. In cases like this, tuple and list should both be handled.